From 3342db32fa783c69188057ac182538fd4277979b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 21 Jun 2018 06:11:28 +0000
Subject: [PATCH] update-kernel: fix install of firware

prevent to exit with error due to set -e, if last firmware does not have
a *.txt file.
---
 update-kernel.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/update-kernel.in b/update-kernel.in
index aaf1cff..4267042 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -252,8 +252,9 @@ find $ROOT/lib/modules -type f -name "*.ko" | xargs modinfo -F firmware | sort -
 	if [ -e "$ROOT/lib/firmware/$FW" ]; then
 		install -pD $ROOT/lib/firmware/$FW $MODLOOP/modules/firmware/$FW
 		# include nvram files if firmware is needed.
-		[ -e "$ROOT/lib/firmware/${FW%.*}.txt" ] && install -pD \
-		    $ROOT/lib/firmware/${FW%.*}.txt $MODLOOP/modules/firmware/${FW%.*}.txt
+		if [ -e "$ROOT/lib/firmware/${FW%.*}.txt" ]; then
+			install -pD $ROOT/lib/firmware/${FW%.*}.txt $MODLOOP/modules/firmware/${FW%.*}.txt
+		fi
 	fi
 done
 
-- 
2.17.1

